packages/office-ui-fabric-react/src/components/DetailsList/DetailsList.tsx

```diff
@@ -398,7 +404,7 @@ export class DetailsList extends BaseComponent<IDetailsListProps, IDetailsListSt
                     <List
                       ref={ this._list }
                       role='presentation'
-                      items={ items }
+                      items={ enableShimmer && !items.length ? this._shimmerInitialItems : items }
```

Comment:
What happens here when the items array is actually just `[]`? Let's say I make an async call, while it's doing that I want to display the shimmer, but then I just get 0 results and want to display an empty list? Will I have to modify the `enableShimmer` prop in that case (make it dependent on the loading status) or can we come up with a better interface? 

Or, as a cheap fix, don't check for empty length but `null`?

Reply:
@cschleiden From my testing with DetailsList it seems that it always needs at least an empty array in order for it to mount. If you provide a null or undefined to items it breaks during the constructor phase and more specific when creating a new Selection on line 123. So checking for null is not an option cause it will never be null. Correct me if I'm wrong...

Reply:
If you think about it shimmer should be dependent on loading status. After all it's a placeholder while waiting for data. The `enableShimmer` right now all is doing is creating an array of 10 null values so that the `onRenderMissingItem` we provide would render 10 shimmer lines. That number it's an average was discussed with designers as we don't know how many items are coming.